solver.md

```diff
@@ -0,0 +1,461 @@
+use and settable solver parameters, is selected in the solver
+configuration file. This file is specified in the input Configuration
+file using a line of the form:
+
+`<Value name="solver_config">../input/solution/cal_solver_config.xml</Value>`
+
+### Syntax
+
+The solver configuration file comprises a series of blocks that
+specify the configuration of the solver to use in each period of the
+model run. For example:  
+
+```
+ <user-configurable-solver year="2005">
+       <solution-tolerance>0.001</solution-tolerance>
+       <solution-floor>0.0001</solution-floor>
+       <calibration-tolerance>0.01</calibration-tolerance>
+       <max-model-calcs>2500</max-model-calcs>
+        .
+        .
+        .
+</user-configurable-solver>
+```
+
+The first line indicates that this is the configuration to be used
+in 2005. If the same solver configuration is to be used for all
+subsequent years, we can add the "fillout" parameter:
+
+`<user-configurable-solver year="2010" fillout="1">`
+
+The first few lines of the configuration specify
+[General Solver Parameters](#General-Solver-Parameters), which apply
+to all solver components. For the most part, these parameters dictate
+things like stopping conditions, which apply to all of the solver
+components that will be used.
+
+Following the general parameters are one or more blocks that specify
+solver components, which direct the solver to run particular solution
+algorithms. For example:  
+
+```
+       <broyden-solver-component>
+           <max-iterations>25</max-iterations>
+           <ftol>1.0e-3</ftol>
+           <solution-info-filter>solvable-nr || (market-type="Tax" &amp;&amp; solvable)</solution-info-filter>
+       </broyden-solver-component>
+```
+
+The first few lines specify parameters that are specific to this
+particular solver component. The `solution-info-filter` line gives a
+[predicate](#Solution-info-filter-Predicates) for determining which
+markets the component will attempt to solve. This filter allows us to
+exclude from a solver component markets that are likely to cause that
+component to fail (for example, markets that make the Jacobian matrix
+singular for the Broyden solver) or whose solution is unlikely
+to be improved by the component.
+
+The algorithms will be run consecutively in the order listed in the
+configuration file. When all have been run, the termination criteria
+will be checked. The sequence of components will be repeated until
+either the model is solved or the maximum number of model evaluations
+has been exceeded.
+
+<a name="General-Solver-Parameters"></a>
+
+### General Solver Parameters
+
+* **calibration-tolerance**: During the calibration periods, the
+  tolerance for determining whether or not the calibration was
+  successful.
+* **max-model-calcs**: The maximum number of times the full model will
+  be evaluated in the course of the solution. If the solver has not
+  found a valid solution by that time, it exits and issues a warning.  
```

Comment:
Not currently, although at least one user has asked for a configuration option to make it so.  The rationale behind why solver failures aren't fatal is that sometimes you get failures in which the only markets out of balance are ones that make tiny contributions to the global picture, and continuing gives users the option to consider the solution "good enough" if the major markets are solved.  Reasonable people might differ over whether that's a sensible way to handle the problem.

Certainly, part of what's lacking is that the solution-floor is kind of a blunt instrument.  We would like to exclude from the solution criteria markets that are so small that it isn't reasonable to insist on meeting the tolerance criteria in relative terms for those markets.  However, what constitutes "small" depends on the units of the output.  Notably, some markets are really "markets" that encapsulate consistency conditions, and their natural scale can be _very_ different from the normal markets that track physical goods.